sdk/keyvault/tests.yml

```diff
@@ -16,5 +16,49 @@ jobs:
       AZURE_TENANT_ID: $(aad-azure-sdk-test-tenant-id)
       AZURE_CLIENT_ID: $(net-keyvault-azure-client-id)
       AZURE_CLIENT_SECRET: $(net-keyvault-azure-client-secret)
-      AZURE_KEYVAULT_URL: $(net-keyvault-azure-keyvault-url)
+      AZURE_KEYVAULT_URL: $[coalesce('$(AZURE_KEYVAULT_NAME)', '$(net-keyvault-azure-keyvault-url)')]
       AZURE_KEYVAULT_TEST_MODE: Live
+
+    PreSteps:
+    - template: ../../eng/pipelines/templates/steps/azure-cli-login.yml
+      parameters:
+        ClientID: $(net-keyvault-azure-client-id)
+        ClientSecret: $(net-keyvault-azure-client-secret)
+        TenantID: $(aad-azure-sdk-test-tenant-id)
+
+    - pwsh: |
+        Write-Host "##vso[task.setvariable variable=AZURE_KEYVAULT_NAME;isOutput=true]${Env:AZURE_KEYVAULT_NAME}"
+      displayName: Generate resource name
+      name: keyvaultvars
+      env:
+        AZURE_KEYVAULT_NAME: $[format('adpnetkv{0:yyyyMMddss}', pipeline.startTime)]
```

Comment:
It may be better to use the build id instead of the time. That will allow us to identify which build the resources came from as well as hopefully make sure we don't accidentally end up with the same name of to parallel jobs (however unlikely).

Reply:
Good idea. I was also going to make different once per-target, i.e. platform/targetFramework. We could get collisions there as well if both are hitting the same vault. I think that's what we've been seeing.